include/klee/KDAlloc/suballocators/sized_regions.h

```diff
@@ -195,43 +195,37 @@ class SizedRegions {
                                       std::size_t const size) const noexcept {
     assert(root && "Cannot compute location_info for an empty treap");
 
-    Node const *currentNode = &*root;
+    Node const *currentNode = root.get();
     Node const *closestPredecessor = nullptr;
-    for (;;) {
-      if (currentNode->getBaseAddress() <= address) {
+    while (currentNode) {
+      if (address < currentNode->getBaseAddress()) {
+        if (address + size > currentNode->getBaseAddress()) {
+          return LocationInfo::LI_Unaligned;
+        }
+        currentNode = currentNode->lhs.get();
+      } else {
         if (address < currentNode->getBaseAddress() + currentNode->getSize()) {
-          if (address + size <=
+          if (address + size >
               currentNode->getBaseAddress() + currentNode->getSize()) {
-            return LocationInfo::LI_Unallocated;
-          } else {
             return LocationInfo::LI_Unaligned;
+          } else {
+            return LocationInfo::LI_Unallocated;
           }
-        } else {
-          closestPredecessor = currentNode;
-          if (!currentNode->rhs) {
-            return {LocationInfo::LI_AllocatedOrQuarantined,
-                    closestPredecessor->getBaseAddress() +
-                        closestPredecessor->getSize()};
-          }
-          currentNode = &*currentNode->rhs;
-        }
-      } else {
-        assert(closestPredecessor &&
-               "If there is no closest predecessor, there must not be a "
-               "predecessor at all. Since regions are only ever split, "
-               "this would mean that the lookup is outside the valid "
-               "range, which has to be handled by the caller.");
-        if (currentNode->getBaseAddress() < address + size) {
-          return LocationInfo::LI_Unaligned;
-        }
-        if (!currentNode->lhs) {
-          return {LocationInfo::LI_AllocatedOrQuarantined,
-                  closestPredecessor->getBaseAddress() +
-                      closestPredecessor->getSize()};
         }
-        currentNode = &*currentNode->lhs;
+        assert(!closestPredecessor || currentNode->getBaseAddress() >
+                                          closestPredecessor->getBaseAddress());
+        closestPredecessor = currentNode;
+        currentNode = currentNode->rhs.get();
       }
     }
+
+    assert(closestPredecessor &&
+           "address must be out of range of the total region, which should "
+           "have been checked by the caller");
```

Comment:
Can you phrase this positively? As in: What does the caller need to ensure?
That's how you did it in `getSize()` and its a lot more concise and easier to understand.